### DISCUSSION TITLE
error uploading reference data to google bucket

### DISCUSSION THREAD

**Discussion by GeorginaAmbriz** [broadinstitute/seqr#3031] [General]
Hello,
I made a local installation of seqr.
I'm trying to upload reference data to google bucket with this command:

GS_BUCKET=gs://exome-1
BUILD_VERSION=38
docker-compose exec pipeline-runner copy_reference_data_to_gs.sh $BUILD_VERSION $GS_BUCKET

but I get this message:
ERROR: No container found for pipeline-runner_1

Thank you so much for any advice you can provide,
Georgina Jimenez Ambriz
Bioinformatics Core
Montreal Neurological Institute-Hospital

**Comment by hanars** [comment]
Hi Georgina

That error means you do not have the pipeline runner image running, which it would be if you had run step 3 of the annotation instruction we provide, and you seem to be on step 4: https://github.com/broadinstitute/seqr/blob/master/deploy/LOCAL_INSTALL.md#annotating-and-loading-vcf-callsets

Please make sure you carefully follow all the steps of the instillation in order and according to the latest documentation. Make sure all steps succeed - if any fails you need to resolve it before trying to proceed to the next step

**Comment by GeorginaAmbriz** [comment]
Hello Hanars,
Thank you so much for answering.
I made all steps to  load a VCF callset. I made steps 1 an 2 and I managed to upload the vcf file in a bucket. The problem begins in step 3.

When I run:
`docker-compose up -d pipeline-runner`

I have the following error message: 
```
elasticsearch is up-to-date

ERROR: for pipeline-runner  Container "bd012ca0c88a" is unhealthy.
ERROR: Encountered errors while bringing up the project.
```

So I don't have a "pipeline-runner" container.

How can I solve this?
Thank you,
Georgina

**Reply by hanars** [reply]
What is the output when you run `docker-compose logs pipeline-runner`

**Reply by GeorginaAmbriz** [reply]
`Attaching to `

**Reply by hanars** [reply]
there should be a full set of logs, not just two words

**Reply by GeorginaAmbriz** [reply]
Hello,
Sorry about the delay.  I had a problem with elasticsearch permissions. I followed a post here: https://github.com/broadinstitute/seqr/discussions/2090, and I did:

```
sudo chown -R 1000:1000 ./data/elasticsearch
sudo chown -R 1000:1000 ./data/postgres
```

This worked to restore seqr local deployment  but I still can't get a pipeline runner image running, I make
`docker-compose up -d pipeline-runner`
And I get this output
```
elasticsearch is up-to-date
Creating geoambriz_pipeline-runner_1 ... 
Creating geoambriz_pipeline-runner_1 ... done

```
But when I run this: 
`docker-compose exec pipeline-runner copy_reference_data_to_gs.sh $BUILD_VERSION $GS_BUCKET`
I got:
`ERROR: No container found for pipeline-runner_1`

tho output of `docker-compose logs -f pipeline-runner`
is:
```
Attaching to geoambriz_pipeline-runner_1
pipeline-runner_1  | exec /entrypoint.sh: exec format error
geoambriz_pipeline-runner_1 exited with code 1
```

**Reply by hanars** [reply]
Can you make sure you are using the latest version of the pipeline runner? Run `docker compose pull pipeline-runner` and see if that fixes it

**Reply by GeorginaAmbriz** [reply]
Hello hanars, 
Thank you for answering. Yes, I have the last version of the pipeline runner, when I run `docker compose pull pipeline-runner` I got this output:
```
Pulling pipeline-runner (gcr.io/seqr-project/pipeline-runner:gcloud-prod)...
gcloud-prod: Pulling from seqr-project/pipeline-runner
Digest: sha256:34543c065ad1e25e1262abadb5b12752693ba486951bc26c9a976c4e14080663
Status: Image is up to date for gcr.io/seqr-project/pipeline-runner:gcloud-prod
```

Still I tried again to get a pipeline_runner image by running `docker-compose up -d pipeline-runner` but I got the same logs:
```
Attaching to geoambriz_pipeline-runner_1
pipeline-runner_1  | exec /entrypoint.sh: exec format error
```

I know that the image is there, this is the output of `docker-compose images`;
```
         Container                             Repository                         Tag         Image Id      Size  
------------------------------------------------------------------------------------------------------------------
elasticsearch                 docker.elastic.co/elasticsearch/elasticsearch   7.16.3        3a5e93284781   583 MB 
geoambriz_pipeline-runner_1   gcr.io/seqr-project/pipeline-runner             gcloud-prod   8971d816e56b   3.09 GB
geoambriz_postgres_1          postgres                                        12.6          145bb83bd157   300 MB 
geoambriz_redis_1             gcr.io/seqr-project/redis                       gcloud-prod   98d2caf140c6   107 MB 
geoambriz_seqr_1              gcr.io/seqr-project/seqr                        gcloud-prod   ed7b116720fa   2.12 GB
```

But it is in State `Exit 1`.
Can this be related to image building architecture? when I run ` docker image inspect 8971d816e56b`, I see that the architecture is
 ```
"Architecture": "arm64",
        "Os": "linux",
```

But my desktop architecture is: `x86_64`

Thank you again for your support.

**Reply by hanars** [reply]
oh yes, this won't work on that machine type. You either need to run this in a VM, or you could build your own version of the docker image. 

To build your own image, you would need to clone the [seqr-loading-pipelines](https://github.com/broadinstitute/seqr-loading-pipelines) repo, and then from the `docker` folder run the following
```
docker build . -f docker/Dockerfile --platform=linux/amd64 -t <tag-name>
```
Then you would update your docker compose to reference `<tag-name> `

**Reply by hanars** [reply]
whatever you want it to be. Generally its your repository name, followed by `:`, followed by a descriptive tag (like `prod` or `latest`)

**Comment by GeorginaAmbriz** [comment]
Hello Hanars,

Thank you so much for your help. 
Building my own pipeline-runner image solved the problem of loading reference data to google bucket.
Following your instructions, I cloned the https://github.com/broadinstitute/seqr-loading-pipelines repo, and  built the image with:
`docker build . -f docker/Dockerfile --platform=linux/amd64 -t pipeline-runner:gcloud-prod`

Then, I updated  pipeline-runner with:
`docker run -d pipeline-runner:gcloud-prod`

I changed the reference to pipeline-runner service in  the `docker-compose.yml` file, by replacing the pipeline-runner image name to my image name:tag  : "pipeline-runner:gcloud-prod". Then I ran the command:
`docker-compose up -d pipeline-runner`

This gave me a  pipeline_runner image running and I could upload the reference data to google bucket.

Now I tried to upload a trio call set, which is already in google bucket. I ran the next commands:
```
GS_BUCKET=gs://exome-1 
BUILD_VERSION=37                 # can be 37 or 38
SAMPLE_TYPE=WES                  # can be WES or WGS
INDEX_NAME=trio1  

INPUT_FILE_PATH=/data/GRCh37/seqr.trio.vcf.gz

docker-compose exec pipeline-runner load_data_dataproc.sh $BUILD_VERSION $SAMPLE_TYPE $INDEX_NAME $GS_BUCKET $INPUT_FILE_PATH
```

but I got this output:
```
+ BUILD_VERSION=37
+ SAMPLE_TYPE=WES
+ INDEX_NAME=family1
+ GS_BUCKET=gs://exome-1
+ INPUT_FILE_PATH=/data/GRCh37/seqr.trio.vcf.gz
+ case ${BUILD_VERSION} in
+ FULL_BUILD_VERSION=GRCh37
+ SOURCE_FILE=gs://exome-1/data/GRCh37/seqr.trio.vcf.gz
+ DEST_FILE=gs://exome-1/data/GRCh37/seqr.mt
+ REFERENCE_DATA_BUCKET=gs://exome-1/reference_data/GRCh37
+ cd /hail-elasticsearch-pipelines/luigi_pipeline
/usr/local/bin/load_data_dataproc.sh: line 27: cd: /hail-elasticsearch-pipelines/luigi_pipeline: No such file or directory
```

I this related to my own image built? 
If it's a different subject, should I open another discussion?
Thank you again

**Comment by hanars** [comment]
I think this is actually a bug with the `load_data_dataproc.sh` script. Try changing the line in the script file from `cd /hail-elasticsearch-pipelines/luigi_pipeline` to `cd /seqr-loading-pipelines/luigi_pipeline` and see if that fixes the issue

**Comment by hanars** [comment]
As an update, it turnsout that it was abug that the platform was not supported, as discussed in https://github.com/broadinstitute/seqr/discussions/3069

This has been fixed, and the latest releases gcloud-prod image has the corect platform and the bug in the loading script it fixed

**Reply by GeorginaAmbriz** [reply]
Hello Hanars,
Thanks for the new release. I deleted my old image and used the new version, it fixed my original problem of uploading the reference genome to gcloud.
Now I have another problem in step 5 of "Annotating and Loading VCF Call Sets", involving my gcloud setup. Since this seems to be a different problem, I'll open another thread if I don't find a direct solution.

Thank you very much for your help.
